SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/AreaManagerNamingViewModel.cs

```diff
@@ -0,0 +1,136 @@
+using SWLOR.Game.Server.Entity;
+using SWLOR.Game.Server.Core.NWScript.Enum;
+using SWLOR.Game.Server.Core.NWScript.Enum.Area;
+using SWLOR.Game.Server.Service;
+using SWLOR.Game.Server.Service.GuiService;
+
+namespace SWLOR.Game.Server.Feature.GuiDefinition.ViewModel
+{
+    public class AreaManagerNamingViewModel : GuiViewModelBase<AreaManagerNamingViewModel, GuiPayloadBase>
+    {
+        public string TemplateAreaNewName
+        {
+            get => Get<string>();
+            set => Set(value);
+        }
+
+        public const int MaxTemplateAreaNewNameLength = 40;
+
+        protected override void Initialize(GuiPayloadBase initialPayload)
+        {
+            TemplateAreaNewName = string.Empty;
+            WatchOnClient(model => model.TemplateAreaNewName);
+        }
+
+        public Action OnClickSubmit() => () =>
+        {
+            if (string.IsNullOrWhiteSpace(TemplateAreaNewName))
+            {
+                return;
+            }
+
+            var templateAreaName = "[Template] " + TemplateAreaNewName;
+
+            if (templateAreaName.Length > MaxTemplateAreaNewNameLength)
+            {
+                SendMessageToPC(Player, $"The template area's new name is to long. Please shorten it to no longer than {TemplateAreaNewName} characters and resubmit.");
+                return;
+            }
+
+            var area = GetArea(Player);
+
+            if (AreaManagerViewModel.GetIsTemplateArea(area))
+            {
+                SendMessageToPC(Player, $"Please do not create a template area of a template area. Go to the area that is part of the core module and template that instead.");
+                Gui.TogglePlayerWindow(Player, GuiWindowType.AreaManagerNaming);
+                return;
+            }
+
+            // Stores transition targets to restore after CreateArea is called.
+            for (var x = GetFirstObjectInArea(area); GetIsObjectValid(x); x = GetNextObjectInArea(area))
+            {
+                if (GetTransitionTarget(x) != OBJECT_INVALID)
+                    SetLocalObject(x, "ORIGINAL_TRANSITION_TARGET", GetTransitionTarget(x));
+            }
+
+            // Generating a random number for the area tag allows for duplicates of the same area without messing up the way in which objects are saved
+            // in each template area. It's very improbable that two duplicate areas would have the exact same tag name and randomize the exact same number.
+            // However, it's less improbable that two areas may have the exact same tag names. For example, if a DM names all his template areas "Ziya's Corellian Event Zone 1" then
+            // there's a high chance all his areas would be "tmp_ziyascorelli" since it caps at 16 characters, which could mess up the ways objects are saved
+            // to each template area. By adding a random 3-digit number to each template area's tag, it makes that duplication highly improbable.
+            var randomNumber = Random(999);
```

Comment:
None of this makes any sense to me.

First of all, tags don't have to be unique. Second of all, randomly picking a number can lead to duplicates - even if it's improbable. And when it does, you're going to run into problems. Third, the concatenation of "tmp_" + randomNumber + GetTag(...) can lead you to a tag longer than 32 characters, which is the maximum.

This is too complicated and doesn't work like you think it does. Reconsider what you're doing here - maybe using something like a template counter would be better. E.G "template_{counter+1}"

Also, even though you shouldn't be picking a random number here - if, in the future, you actually do need something which requires RNG, please use the Random service provided in SWLOR rather than the NWN one, as you're doing here.